src/artemis/system_tests/test_fgs_plan.py

```diff
@@ -0,0 +1,142 @@
+        omega_start=0.0,
+        omega_increment=0.1,
+        num_images=50,
+        use_roi_mode=False,
+        run_number=0,
+        det_dist_to_beam_converter_path="src/artemis/devices/unit_tests/test_lookup_table.txt",
+    )
+    eiger = EigerDetector(
+        detector_params=detector_params, name="eiger", prefix="BL03S-EA-EIGER-01:"
+    )
+
+    # Otherwise odin moves too fast to be tested
+    eiger.cam.manual_trigger.put("Yes")
+
+    # S03 currently does not have StaleParameters_RBV
+    eiger.wait_for_stale_parameters = lambda: None
+    eiger.odin.check_odin_initialised = lambda: (True, "")
+
+    yield eiger
+
+
+params = FullParameters()
+params.beamline = SIM_BEAMLINE
+
+
+@pytest.fixture
+def RE():
+    return RunEngine({})
+
+
+@pytest.fixture
+def fgs_composite():
+    fast_grid_scan_composite = FGSComposite(
+        insertion_prefix=params.insertion_prefix,
+        name="fgs",
+        prefix=params.beamline,
+    )
+    return fast_grid_scan_composite
+
+
+@pytest.mark.s03
+@patch("bluesky.plan_stubs.wait")
+@patch("bluesky.plan_stubs.kickoff")
+@patch("bluesky.plan_stubs.complete")
+def test_run_gridscan(
+    complete: MagicMock,
+    kickoff: MagicMock,
+    wait: MagicMock,
+    eiger: EigerDetector,
+    RE: RunEngine,
+    fgs_composite: FGSComposite,
+):
+
+    fgs_composite.wait_for_connection()
+    # Would be better to use get_plan instead but eiger doesn't work well in S03
+    RE(run_gridscan(fgs_composite, eiger, params))
+
+
+@pytest.mark.s03
+def test_read_hardware_for_ispyb(
+    eiger: EigerDetector,
+    RE: RunEngine,
+    fgs_composite: FGSComposite,
+):
+
+    undulator = fgs_composite.undulator
+    synchrotron = fgs_composite.synchrotron
+    slit_gaps = fgs_composite.slit_gaps
+
+    @bpp.run_decorator()
+    def read_run(u, s, g):
+        yield from read_hardware_for_ispyb(u, s, g)
+
+    fgs_composite.wait_for_connection()
+    RE(read_run(undulator, synchrotron, slit_gaps))
+
+
+@pytest.mark.s03
+@patch("bluesky.plan_stubs.wait")
+@patch("bluesky.plan_stubs.kickoff")
+@patch("bluesky.plan_stubs.complete")
+@patch("artemis.fgs_communicator.FGSCommunicator")
+@patch("artemis.fast_grid_scan_plan.tidy_up_plans")
+@patch("artemis.fast_grid_scan_plan.run_gridscan_and_move")
+def test_full_plan_tidies_at_end(
+    run_gridscan_and_move: MagicMock,
+    tidy_plans: MagicMock,
+    communicator: MagicMock,
+    complete: MagicMock,
+    kickoff: MagicMock,
+    wait: MagicMock,
+    eiger: EigerDetector,
+    RE: RunEngine,
+    fgs_composite: FGSComposite,
+):
+
+    RE(get_plan(params, communicator))
+    tidy_plans.assert_called_once()
+
+
+@pytest.mark.s03
+@patch("bluesky.plan_stubs.wait")
+@patch("bluesky.plan_stubs.kickoff")
+@patch("bluesky.plan_stubs.complete")
+@patch("artemis.fgs_communicator.FGSCommunicator")
+@patch("artemis.fast_grid_scan_plan.tidy_up_plans")
+@patch("artemis.fast_grid_scan_plan.run_gridscan_and_move")
+def test_full_plan_tidies_at_end_when_plan_fails(
```

Comment:
Should: Both these tests fail with due to an import issue

Reply:
Also, given we're running these as system tests anyway do we need to mock out the `tidy_up_plans`? And if we're running with `fake_zocalo` and the dev database we probably don't need to mock out the communicator either? My general philosophy with the tests is that for unit test we mock out lots so that we're just testing the thing we want to test as unit tests are quick and good to have lots of. But for system tests is the opposite we want to test as much as possible because they tend to be slower and we probably only want a few of them

Reply:
communicator hasn't existed for a few revisions now already. fixed version uses the real callbacks.